ipns/name.go

```diff
@@ -0,0 +1,94 @@
+package ipns
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/multiformats/go-multibase"
+)
+
+// Name represents an [IPNS Name].
+//
+// [IPNS Name]: https://specs.ipfs.tech/ipns/ipns-record/#ipns-name
+type Name struct {
```

Comment:
:thought_balloon: Crazy idea: would it be easier to reason if we had  `peer.ID` and   `ipns.ID`? or worse? :upside_down_face: 
We could clean up naming in spec too, if needed. (Fine to keep `Name`, just thought if we want to change the term, it is good time, once people start using `ipns.Name` it will be difficult to change)

```suggestion
type ID struct {
```

Reply:
I think `ID` could be a great idea, especially considering we're removing `/ipns` from the `.String`. The specification, as it is now, explicitly says we should include `/ipns` in its string format.

Reply:
We I've added that sentence to specs as best practice, if its confusing, we can remove it or make it more clear:


```diff
- A good practice is to prefix IPNS Name with /ipns/ namespace, and refer to IPNS addresses as /ipns/{ipns-name} (or /ipns/{libp2p-key}).
+ In context outside of IPNS, when other content identifier are used, a good practice is to prefix IPNS Name with /ipns/ namespace, and refer to IPNS addresses as /ipns/{ipns-name} (or /ipns/{libp2p-key}).

Reply:
Yes, we can update the spec to make it clearer, and keep `Name` with that name (perhaps pun intended).